src/lbaf/IO/lbsMeshBasedVisualizer.py

```diff
@@ -92,20 +92,42 @@ def __init__(
                 else 0.0 for d in range(3)]
             for i in self.__phases[0].get_object_ids()}
 
-        # Initialize maximum edge volume
-        self.__max_object_volume = 0.0
+        # Initialize maximum object atrribute values
+        self.__object_load_max = 0.0
+        self.__object_volume_max = 0.0
 
-        # Compute object load range
-        self.__load_range = [math.inf, 0.0]
+        # Compute space-time object QOI range across all objects in phase
+        oq_min, oq_max, oq_all, store_all = math.inf, -math.inf, set(), True
         for p in self.__phases:
-            for r in p.get_ranks():
-                for o in r.get_objects():
-                    # Update load range when necessary
-                    load = o.get_load()
-                    if load > self.__load_range[1]:
-                        self.__load_range[1] = load
-                    if load < self.__load_range[0]:
-                        self.__load_range[0] = load
+            for o in p.get_objects():
+                # Update maximum object load as needed
+                if (ol := o.get_load()) > self.__object_load_max:
+                    self.__object_load_max = ol
+
+                # Retain all QOI values while support remains small
+                oq = getattr(o, f"get_{object_qoi}")()
+                if store_all:
+                    oq_all.add(oq)
+                    if len(oq_all) > 20:
+                        # Do not store QOI values if support is too large
+                        oq_all = None
+                        store_all = False
+
+                # Update extrema
+                if oq < oq_min:
+                    oq_min = oq
+                if oq > oq_max:
+                    oq_max = oq
+
+        # Store either range or support
+        if store_all:
+            self.__object_qoi_range = oq_all
+            self.__logger.info(
+                f"\t{self.__object_qoi} has {len(self.__object_qoi_range)} distinct values")
+        else:
+            self.__object_qoi_range = (oq_min, oq_max)
+            self.__logger.info(
+                f"\t{self.__object_qoi} range: [{self.__object_qoi_range[0]:.4g}; {self.__object_qoi_range[1]:.4g}]")
```

Comment:
`__init__` is now over 200 lines long. Can you extract this into a separate method?